### ISSUE TITLE
Make nf-tests go faster

### ISSUE THREAD

**Issue by nschcolnicov** [nf-core/smrnaseq#389]
Not sure about the priority of this ticket, or if we have enough time to tackle it, but nf-tests run in an OK time when running in CI because they run in parallel, but when having to update snapshots we have to run the sequentially, making them take around 1h. The scope of this ticket is to assess and implement (if possible) the following ideas for making them go faster:

- Update tests to use bowtie index for faster runs
- Subset test_full_filter_contamination test profile for running in CI nf-tests
- Subset all nf-test input files

FYI @grst @atrigila @apeltzer

**Comment by apeltzer** [comment]
+1 for using the bowtie indices, that makes sense 👍

**Comment by apeltzer** [comment]
@atrigila  suggested to run on not all 8 samples in the samplesheet but rather reduce to 3-4 maybe.

**Comment by nschcolnicov** [comment]
We will implement the following changes:

- Reduce samplesheet size for all test cases, whenever possible.
- Remove cleanup parameter from all tests, not doing anything.
- Nice to have: Have bowtie index as input also for hairpin and mature fasta files, should we move it to igenomes?

Combine the following profiles into one (we saw that they were running redundant modules and subworkflows)
 
- test and test_index: Combine and add bowtie_index as input
- test_technical_repeats, test_no_genome and test_skipfastp

Dont combine these tests
- test_umi : Add bowtie_index as input
- test_mirgenedb

**Comment by nschcolnicov** [comment]
https://github.com/nf-core/smrnaseq/pull/394
